training/src/test/scala/com/airbnb/aerosolve/training/TrainingTestHelper.scala

```diff
@@ -47,4 +44,22 @@ object TrainingTestHelper {
     }
     (examples, label, numPos)
   }
+
+  def makeRegressionExamples = {
+    val examples = ArrayBuffer[Example]()
+    val label = ArrayBuffer[Double]()
+    val rnd = new java.util.Random(1234)
+
+    for (i <- 0 until 200) {
+      val x = 4.0 * (2.0 * rnd.nextDouble() - 1.0)
+      val y = 4.0 * (2.0 * rnd.nextDouble() - 1.0)
+
+      val quadradtic = x * x - 2 * y * y - 0.5 * x + 0.2 * y
```

Comment:
Changed the curve to have some flat regions and added a test that the algorithm does the right thing in the associated regions.